### DISCUSSION TITLE
请问一下在基于 FastAPI 的 Web API的下游任务里面，如何只是用CPU进行部署请求？

### DISCUSSION THREAD

**Discussion by dhr613** [magicpdf/Magic-PDF#1235] [Q&A]
docker run -itd --name=mineru_server -p 8888:8000 quincyqiang/mineru:0.1-models（我把gpu的参数去掉了）
然后请求体是：
curl -X 'POST' \
  'http://localhost:8888/pdf_parse?parse_method=auto&is_json_md_dump=true' \
  -H 'accept: application/json' \
  -H 'Content-Type: multipart/form-data' \
  -F 'pdf_file=@"/mnt/d/dhr/tests/datas/_lwq.pdf";type=application/pdf' \
  -v

依然会显示：
{"error":"Found no NVIDIA driver on your system. Please check that you have an NVIDIA GPU and installed a driver from http://www.nvidia.com/Download/index.aspx"}

**Comment by bhaswata08** [comment]
If you plan to use cpus
execute this inside the docker container

sed -i 's|cuda|cpu|g' /root/magic-pdf.json

If you plan to use GPUS
`docker run --gpus all -itd --name=mineru_server -p 8888:8000 quincyqiang/mineru:0.1-models`

Make sure you have Nvidia-container toolkit installed on your system for gpu passthrough https://docs.nvidia.com/datacenter/cloud-native/container-toolkit/latest/install-guide.html
